controller/service.go

```diff
@@ -118,10 +127,19 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.client.Errorf(svc, "LoadBalancerFailed", "invalid requested loadbalancer IPs: %s", err)
 			return true
 		}
-		if len(desiredLbIPs) > 0 && !isEqualIPs(lbIPs, desiredLbIPs) {
-			level.Info(l).Log("event", "clearAssignment", "reason", "differentIPRequested", "msg", "user requested a different IP than the one currently assigned")
-			c.clearServiceState(key, svc)
-			lbIPs = []net.IP{}
+		if len(desiredLbIPs) > 0 {
+			desiredPool, err := c.ips.GetPoolOwner(desiredLbIPs)
+			if err != nil {
+				level.Error(l).Log("event", "loadbalancerIP", "error", err, "msg", "invalid ip pool for requested loadbalancer IPs")
+				c.client.Errorf(svc, "LoadBalancerFailed", "invalid ip pool for requested loadbalancer IPs: %s", err)
+				c.clearServiceState(key, svc)
+				return true
+			}
+			if len(lbIPs) != len(desiredLbIPs) || (!isEqualIPs(lbIPs, desiredLbIPs) && !desiredPool.DisablePoolSelection) {
```

Comment:
I  think checking the disable selection is not enough here. We need to chceck if the pool is not suitable for the service we are looking at (checking the selectors)